### PR TITLE
#12: Add tests for violation codes and docs

### LINKED ISSUE #12: Test violations codes and docs
We need to tests that all violations 

1. have unique codes
2. correctly sorted inside the code
3. have docs in the module docstrings

You can have a look at the example (this code can be even copy-pasted): https://github.com/wemake-services/wemake-python-styleguide/tree/master/tests/test_violations

### PR DESCRIPTION
**Pull request by nifadyev**
# I have made things!

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [ ] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues

Closes #12
Based on PR #88

### CODE REVIEW

File: tests/test_violations/test_codes.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from collections import Counter
```
Comment (nifadyev) [inline tests/test_violations/test_codes.py:1]: Should [these](https://github.com/wemake-services/wemake-python-styleguide/blob/master/tests/test_violations/test_codes.py#L26-L45) tests be adopted as well? `all_module_violations` mapping is not sorted and violation codes are not sequential

See `all_violation_codes` output
```
{
      <module 'dotenv_linter.violations.parsing' from '/home/outreach/Projects/dotenv-linter/dotenv_linter/violations/parsing.py'>: {
          1: <class 'dotenv_linter.violations.parsing.ParsingViolation'>,
      },
      <module 'dotenv_linter.violations.assigns' from '/home/outreach/Projects/dotenv-linter/dotenv_linter/violations/assigns.py'>: {
          200: <class 'dotenv_linter.violations.assigns.SpacedAssignViolation'>,
      },
      <module 'dotenv_linter.violations.comments' from '/home/outreach/Projects/dotenv-linter/dotenv_linter/violations/comments.py'>: {
          400: <class 'dotenv_linter.violations.comments.SpacedCommentViolation'>,
      },
      <module 'dotenv_linter.violations.names' from '/home/outreach/Projects/dotenv-linter/dotenv_linter/violations/names.py'>: {
          100: <class 'dotenv_linter.violations.names.SpacedNameViolation'>,
          101: <class 'dotenv_linter.violations.names.IncorrectNameViolation'>,
          102: <class 'dotenv_linter.violations.names.DuplicateNameViolation'>,
          103: <class 'dotenv_linter.violations.names.RawNameViolation'>,
          104: <class 'dotenv_linter.violations.names.ReservedNameViolation'>,
      },
      <module 'dotenv_linter.violations.values' from '/home/outreach/Projects/dotenv-linter/dotenv_linter/violations/values.py'>: {
          300: <class 'dotenv_linter.violations.values.SpacedValueViolation'>,
          301: <class 'dotenv_linter.violations.values.QuotedValueViolation'>,
      },
  }
```

File: tests/test_violations/test_implementation.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import pytest
+
+from dotenv_linter.grammar.parser import DotenvParser
+from dotenv_linter.violations.base import BaseFSTViolation, BaseViolation
+
+
+def test_visitor_returns_location() -> None:
```
Comment (nifadyev) [inline tests/test_violations/test_implementation.py:7]: Not sure, what is the purpose of this. Maybe test method `as_line()`?

Comment (sobolevn) [reply tests/test_violations/test_implementation.py:7]: This file can be safely removed, I think

**Review by sobolevn — COMMENTED** [review]
Thank you!

File: tests/conftest.py [hunk 3]
```diff
@@ -11,6 +15,8 @@
     BaseViolation,
 )
 
+ALL_VIOLATIONS_TYPE: TypeAlias = Dict[ModuleType, List[BaseViolation]]
```
Comment (sobolevn) [inline tests/conftest.py:18]: `ALL_VIOLATIONS_TYPE` is a constant.
While `AllViolationsType` is a type :)

Please, rename it.

**Review by sobolevn — COMMENTED** [review]
Thanks! Just one nitpick left

**Comment by nifadyev** [comment]
@sobolevn , could you please help with passing the test `test_all_violations_are_final` on python 3.9-3.10. Docs on [final](https://docs.python.org/3/library/typing.html#typing.final) decorator says that attribute `__final__` is available since 3.11. It is the reason the test fails. But cannot find the reason the same test with same python 3.9 passes in `wamake-python-styleguide`
i've tried import `final` from `typing_extensions`, no luck

**Comment by sobolevn** [comment]
I use `typing_extensions` for `final` import. https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/violations/best_practices.py#L176

But, make sure that `typing_extensions` has the latest version.

**Review by sobolevn — APPROVED** [review]
Thank you!

**Pull request merged**